app/assets/javascripts/initializers/initializeAllFollowButts.js

```diff
@@ -1,10 +1,54 @@
 function initializeAllFollowButts() {
   var followButts = document.getElementsByClassName('follow-action-button');
   for (var i = 0; i < followButts.length; i++) {
-    initializeFollowButt(followButts[i]);
+    if (!followButts[i].className.includes("follow-user")) {
+      initializeFollowButt(followButts[i]);
+    };
   }
 }
 
+function fetchUserFollowStatuses(idButtonHash) {
+  var idString = "";
+  Object.keys(idButtonHash).forEach(function(id) {
+    idString += ("ids[]=" + id + "&");
+  });
+
+  fetch(`/follows/bulk_show?${idString}followable_type=User`, {
+    method: 'GET',
+    headers: {
+      Accept: 'application/json',
+      'X-CSRF-Token': window.csrfToken,
+      'Content-Type': 'application/json',
+    },
+    credentials: 'same-origin',
+  }).then((response) => response.json())
+    .then((idStatuses) => {
+      Object.keys(idStatuses).forEach(function(id) {
+        addButtClickHandle(idStatuses[id], idButtonHash[id]);
+      })
+    });
+}
+
+function initializeUserFollowButtons(buttons) {
+  var userIds = {};
+  for (var i = 0; i < buttons.length; i++) {
```

Comment:
not necessary but a tip: `Array.from(buttons)` lets you iterate using `forEach` as well, but the for is likely faster here